pkg/gameservers/controller_test.go

```diff
@@ -155,6 +156,105 @@ func TestControllerSyncGameServer(t *testing.T) {
 	})
 }
 
+func TestControllerSyncGameServerWithInitSidecar(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Creating a new GameServer", func(t *testing.T) {
+		c, mocks := newFakeController()
+		updateCount := 0
+		podCreated := false
+		fixture := &agonesv1.GameServer{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+			Spec: agonesv1.GameServerSpec{
+				Ports: []agonesv1.GameServerPort{
+					{ContainerPort: 7777},
+					{ContainerPort: 8888, Container: ptr.To("sidecar")},
```

Comment:
well that's a handy package I didn't know about! `k8s.io/utils/pt` going to steal that.